lua/tests/spec/component_spec.lua

```diff
@@ -241,6 +242,57 @@ describe('Fileformat component', function()
   end)
 end)
 
+describe('Filetype component', function()
+  local filetype
+  -- local synIDattr
+
+  before_each(function()
+    filetype = vim.bo.filetype
+    vim.bo.filetype = 'lua'
+
+    -- synIDattr = vim.fn.synIDattr
+    -- vim.fn.synIDattr = function() return '#000' end
+  end)
+
+  after_each(function()
+    vim.bo.filetype = filetype
+    -- vim.fn.synIDattr = synIDattr
+  end)
+
+  it('does not add icon when library unavailable', function()
+    local opts = build_component_opts({
+      component_separators = {'', ''},
+      padding = 0
+    })
+    assert_component('filetype', opts, 'lua')
+  end)
+
+  it('colors nvim-web-devicons icons', function()
+    package.loaded['nvim-web-devicons'] = {
+      get_icon = function(_, _)
+        -- return icon, icon_highlight_group
+        return '*', 'test_highlight_group'
+      end
+    }
+
+    -- local mock_hl = mock(hl, true)
+    -- mock_hl.create_component_highlight_group.returns('MyCompHl')
+
+    local hl = require 'lualine.highlight'
+    stub(hl, 'create_component_highlight_group')
+    hl.create_component_highlight_group.returns('MyCompHl')
+
+    local opts = build_component_opts({
+      component_separators = {'', ''},
+      padding = 0
+    })
+    -- assert.stub(hl.create_component_highlight_group).was_called_with(
+    --   {fg = '#000'}, 'test_highlight_group', {}
+    -- )
```

Comment:
Can't seem to get this assertion to work. Left what I tried as comments in case anyone has any ideas. Will remove if unsuccessful.